### PR TITLE
Add main page for API documentation

### PR DESCRIPTION
**Pull request by pyohannes**
This PR adds a main page for the API reference documentation.

It also makes sure the version in the API documentation is updated by our release build scripts.

### CODE REVIEW

**Review by lalitb — APPROVED** [review]
LGTM. Thanks for adding the initial document. Probably, we should be able to extend it further through #571 . Also, not related to this PR, but we would be needing Doxygen doc generation for resources and samplers (and few others ? ) from SDK. I will create a ticket for that.

**Comment by pyohannes** [comment]
> Also, not related to this PR, but we would be needing Doxygen doc generation for resources and samplers (and few others ? ) from SDK.

Yes, but I think those should be separate. Instrumenters will only be interested in the API, and I think it would be confusing to have SDK parts mixed in there.

File: api/docs/Overview.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+enables developers to instrument applications and libraries with the
+OpenTelemetry C++ API while being completely agnostic of how telemetry data is
+exported and processed.
+
+## Library design
+
+The OpenTelemetry C++ API is provided as a header-only library and supports all
+recent versions of the C++ standard, down to C++11.
+
+A single application might dynamically or statically link different libraries
```
Comment (ThomsonTan) [inline api/docs/Overview.md:17]: nit: perhaps `link to`?

File: api/docs/Overview.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+that were compiled with different compilers, while several of the linked
+libraries are instrumented with OpenTelemetry. OpenTelemetry C++ supports those
+scenarios by providing a stable ABI. This is achieved by a careful API design,
+and most notably by providing ABI stable versions of classes from the standard
+library. All those classes are provided in the `opentelemetry:nostd` namespace.
```
Comment (ThomsonTan) [inline api/docs/Overview.md:22]: nit
```suggestion
library. All those classes are provided in the `opentelemetry::nostd` namespace.
```

**Review by ThomsonTan — APPROVED** [review]

**Review by jsuereth — APPROVED** [review]

**Pull request merged**